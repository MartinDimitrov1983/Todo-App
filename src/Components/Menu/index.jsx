import React from 'react';
import PropTypes from 'prop-types';
import { Button, AddTodo } from '..';
import { RESOLVE_ALL, UNRESOLVE_ALL, REMOVE_ALL } from '../../Constants';
import styles from './index.module.css';

const Menu = ({
  resolveHandler,
  unResolveHandler,
  deleteHandler,
  addHandler,
  onInputChangeHandler,
  value,
}) => {
  return (
    <aside className={styles.menu}>
      <div className={styles.buttons}>
        <Button text={RESOLVE_ALL} onClick={resolveHandler}></Button>
        <Button text={UNRESOLVE_ALL} onClick={unResolveHandler}></Button>
        <Button text={REMOVE_ALL} onClick={deleteHandler}></Button>
      </div>
      <AddTodo
        onClick={addHandler}
        onInputChange={onInputChangeHandler}
        value={value}
      />
    </aside>
  );
};

Menu.propTypes = {
  resolveHandler: PropTypes.func,
  unResolveHandler: PropTypes.func,
  deleteHandler: PropTypes.func,
  addHandler: PropTypes.func,
  onInputChangeHandler: PropTypes.func,
  value: PropTypes.string,
};

export default Menu;
