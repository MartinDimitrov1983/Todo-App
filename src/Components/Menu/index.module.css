.menu {
    display: flex;
    flex-direction: column;
    min-width: 200px;
    align-items: center;
    padding: 20px;
    padding-top: 50px;
    background: rgb(238 243 243);
    filter: drop-shadow(0 0 0.2rem rgb(193, 193, 193));
    /* min-height: 100vh;
    align-self: stretch; */

  }

  .buttons {
    display: flex;
    gap: 10px;
    flex-direction: column;
    max-width: 100px;
  }