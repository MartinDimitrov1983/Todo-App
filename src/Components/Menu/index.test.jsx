import React from 'react';
import { render } from '@testing-library/react';

import Menu from './index';

test('Snapshot Menu', () => {
  const { asFragment } = render(
    <Menu
      resolveHandler={() => {}}
      unResolveHandler={() => {}}
      deleteHandler={() => {}}
      addHandler={() => {}}
      onInputChangeHandler={() => {}}
      value="Test"
    />,
  );
  expect(asFragment()).toMatchSnapshot();
});
