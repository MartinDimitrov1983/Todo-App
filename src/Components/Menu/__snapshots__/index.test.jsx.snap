// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Menu 1`] = `
<DocumentFragment>
  <aside
    class="menu"
  >
    <div
      class="buttons"
    >
      <button
        class="button"
        type="button"
      >
        <div>
          Resolve All
        </div>
      </button>
      <button
        class="button"
        type="button"
      >
        <div>
          Unresolve All
        </div>
      </button>
      <button
        class="button"
        type="button"
      >
        <div>
          Remove All
        </div>
      </button>
    </div>
    <div
      class="container"
    >
      <input
        class="input"
        placeholder="Add Todo"
        value="Test"
      />
      <button
        class="button"
        type="button"
      >
        <div>
          Add Todo
        </div>
      </button>
    </div>
  </aside>
</DocumentFragment>
`;
